OpenSim/Sandbox/ExampleHopperDevice/exampleHopperDevice.cpp

```diff
@@ -1,428 +1,285 @@
-we'll make it a type of ModelComponent. Since Components can be composed
-by subcomponents by their nature, we don't need to implement any additional
-functionality. Later we will add methods to evaluate the performance of the
-device. This will be the scaffold for our device, so we'll call the class,
-Device. */
-class Device : public ModelComponent {
-    OpenSim_DECLARE_CONCRETE_OBJECT(Device, Component);
-
-public:
-    /** Add outputs so we can report device quantities we care about. */
-    /** The length of the device from anchor to anchor point. */
-    OpenSim_DECLARE_OUTPUT(length, double, getLength, SimTK::Stage::Position);
-
-    // TODO: add other outputs.
-
-    /** Member functions to access values of interest from the device. */
-    double getLength(const SimTK::State& s) const {
-        return getComponent<PathActuator>("cableAtoB").getLength(s);
-    }
-
-    // TODO: add other output functions.
-
-protected:
-    /** Optionally change the color of the device's actuator path
-        as its tension changes. */
-    void extendRealizeDynamics(const SimTK::State& s) const override {
-    }
-}; // end Device
-
-/**
-* Create a Controller that produces a control signal = k * a, where `k` is
-* the gain property, and `a` is the activation input. This is intended to model
-* proportional myoelectric device controllers. This Controller can control any
-* ScalarActuator. The ScalarActuator that this control controls is set using
-* the `device` connector.
-* http://en.wikipedia.org/wiki/Proportional_Myoelectric_Control
-*/
-class PropMyoController : public OpenSim::Controller {
-    OpenSim_DECLARE_CONCRETE_OBJECT(PropMyoController, OpenSim::Controller);
-public:
-
-    // TODO: gain property
-
-    // TODO: actuator connector
-
-    // TODO: myo_control output
-
-    OpenSim_DECLARE_INPUT(activation, double, SimTK::Stage::Model,
-            "The activation signal that this controller's signal is "
-            "proportional to.");
-
-    PropMyoController() {
-        constructProperties();
-    }
-
-    double computeControl(const SimTK::State& s) const {
-        // Compute the proportional control of GAIN * activation (Input)
-        // TODO
-        return 0;
-    }
-
-    void computeControls(const SimTK::State& s,
-        SimTK::Vector& controls) const override {
-        double signal = computeControl(s);
-        // Add in this control signal to controls.
-        // TODO
-        // const auto& actuator = getConnectee<Actuator>("actuator");
-        SimTK::Vector thisActuatorsControls(1, signal);
-        // Add in this controller's controls for the actuator
-        // TODO
-    }
+//TODO: Provide the name of the output corresponding to the hopper's height.
+//      Hint: the hopper's pelvis is attached to ground with a vertical slider
+//      joint. Also see Device::getHeight() in defineDeviceAndController.h.
+// [Step 3, Task B]
+static const std::string hopperHeightOutput{"/Dennis/?????"}; //fill this in
```

Comment:
This needs to be filled in earlier (for Step 1, Task A). May also want to refer to this in the comments for that part.